client/src/pages/EnergiserDetail.js

```diff
@@ -0,0 +1,87 @@
+import React from "react";
+import Navbar from "./components/Navbar";
+import Footer from "./components/Footer";
+import { BsStar } from "react-icons/bs";
+import { BsStarHalf } from "react-icons/bs";
+import { BsStarFill } from "react-icons/bs";
+import { FaRegHeart } from "react-icons/fa";
+import { FaShare } from "react-icons/fa";
+import { useParams } from "react-router-dom";
+import { useState, useEffect } from "react";
+import "./EnergiserDetail.css";
+
+function EnergiserDetail() {
+    const { id } = useParams();
+    const [item, setItem] = useState(null);
+
+    useEffect(() => {
+        fetch(`/api/energizer-detail/${id}`)
+            .then((res) => {
+                if (!res.ok) {
+                    throw new Error(res.statusText);
+                }
+                return res.json();
+            })
+            .then((body) => {
+                setItem(body);
+            })
+            .catch((err) => {
+                console.error(err);
+            });
+    }, [id]);
+
+    const ShareButton = () => {
+        const handleShare = () => {
+            const shareUrl = window.location.href;
+            navigator.clipboard.writeText(shareUrl);
+            window.alert("Link copied to clipboard!");
+        };
+
+        return (
+            <div className="share-sec">
+                <button className="share-btn" onClick={handleShare}>
+                    <FaShare />
+                </button>
+                <p>Share with others</p>
+            </div>
+        );
+    };
+
+    if (!item) {
+        return <div>Loading...</div>;
+    }
+
+    return (
+        <main className="main-page">
+            <Navbar />
```

Comment:
Could these elements (main, Navbar, Footer, etc) be put at a higher level in component hiearchy so we are not repeating them on every individual page?

Reply:
yes It can Bertil and its a good ldea I will look into it